jdbc/sqlserver/src/main/java/com/google/cloud/sql/sqlserver/SocketFactory.java

```diff
@@ -46,21 +48,23 @@ public class SocketFactory extends javax.net.SocketFactory {
    * connections to a Cloud SQL instance.
    */
   public SocketFactory(String socketFactoryConstructorArg) throws UnsupportedEncodingException {
-    String[] s = socketFactoryConstructorArg.split("\\?");
-    this.props.setProperty(CoreSocketFactory.CLOUD_SQL_INSTANCE_PROPERTY, s[0]);
-    if (s.length == 2 && s[1].length() > 0) {
-      String[] queryParams = s[1].split("&");
+    List<String> s = Splitter.on('?').splitToList(socketFactoryConstructorArg);
```

Comment:
Do we have any unit tests covering this code?

We should ensure we have thorough testing of this code before changing it here, just out of caution. Also, once we have those tests in place, we should clean this code up -- on the surface it looks more complicated than it needs to be.

Reply:
Yes, we have unit tests for this code in [JdbcSqlServerUnitTests.java](https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/main/jdbc/sqlserver/src/test/java/com/google/cloud/sql/sqlserver/JdbcSqlServerUnitTests.java).

I will create a new issue to refactor this code.